src/pages/en/core-concepts/astro-components.md

```diff
@@ -125,22 +125,45 @@ They can be used to style your components, and all style rules are automatically
 
 ### Client-Side Scripts
 
-To send JavaScript to the browser without [using a framework component](/en/core-concepts/component-hydration) (React, Svelte, Vue, Preact, Solid...) you can use a `<script>` tag in your Astro component body.
+To send JavaScript to the browser without [using a framework component](/en/core-concepts/framework-components) (React, Svelte, Vue, Preact, SolidJS, AlpineJS, Lit...) you can use a `<script>` tag in your Astro component template and send JavaScript to the browser that executes in the global scope.
+
 
 ```astro
----
-// Your component script here!
----
 <script>
-  document.querySelector('h1').style.color = 'red';
+  // Will be rendered into the HTML exactly as written!
+  // ESM imports will not be resolved relative to the file.
 </script>
 
-<h1>Hello, world!</h1>
+<script hoist type="module">
```

Comment:
`type="module"` shouldn't be doing anything here, but it also can't hurt. I might be missing the latest guidance on this, so if this feels right to you we can leave it in and then revisit post-merge

Reply:
I think the reasoning for this was that if you import anything you _do_ need `type="module"` for it to work and people often miss that. (At least that's my understanding from hacking around — might be wrong.)